// WebService/test/DAO/TestBean.ts
import { Facility } from 'CommonUtils';

const Extends = Facility.Extends;

/*
function Do() {
    setTimeout(function () {
        TestIO.emit('data', {
            ID: '54321',
            Name: 'Hello World',
            Time: new Date(),
        });
    }, 100);
}

function Do1() {
    TestIO.emit('data', {
        ID: '54321',
        Name: 'Hello World',
        Time: new Date(),
    });
}


export const TestBean = Extends(require('WebService').BaseDAO, {
    Get: function (context) {
        TestIO = new (require('events').EventEmitter)();
        TestIO.on('data', context.Regist(this.DataReady));
        Do1.apply(TestIO);

        //context.Result = {
        //    ID : '123456',
        //    Name : 'Hello World',
        //};
    },

    Post: function (context) { },

    DataReady: function (obj: any) {
        this.Result = obj;
    }
});


export const DbTestBean = Extends(require('WebService').BaseDAO, {
    async Get(id: string) {
        return {
            ID: id,
            Name: 'Hello World',
            Time: new Date(),
        };
    },

    Set(obj: any, opts: any) {
        console.log(obj);
        return 'OK';
    },

});
*/
