// DataAccess/index.ts
import { IDBAccessor } from './lib/DbAccessor'

import { PgAccessor } from './lib/PgAccessor';
import { MrAccessor } from './lib/MrAccessor';
import { MaAccessor } from './lib/MaAccessor';

export { QueryFlags } from './lib/QueryOptions';
export { PgAccessor } from './lib/PgAccessor';
export { MaAccessor } from './lib/MaAccessor';
export { MrAccessor } from './lib/MrAccessor';
export { LevelStore } from './lib/LevelStore';
export { LMStore, DBContext } from './lib/LMStore';
export { Entities } from './lib/Entities';

export function Create(opts: any): IDBAccessor {
    let Result = null;

    let provider = opts.provider || 'mysql';
    if (provider) {
        switch (provider) {
            case 'maria':
                Result = new MaAccessor(opts);
                break;

            case 'mysql':
                Result = new MrAccessor(opts);
                break;

            case 'postgres':
                Result = new PgAccessor(opts);
                break;
        }
    }

    if (!Result) {
        throw Error(`No Databse Provider: ${provider}`);
    }

    return Result;
}
